src/Core/Main.vala

```diff
@@ -2385,9 +2385,13 @@ public class Main : GLib.Object{
 			sh += "echo '" + _("Updating GRUB menu...") + "' \n";
 			
 			if (target_distro.dist_type == "redhat"){
-				sh += "%s grub2-mkconfig -o /boot/grub2/grub.cfg \n".printf(chroot);
+				if (target_distro.is_efi){
+					sh += "%s grub2-mkconfig -o /boot/efi/EFI/%s/grub.cfg \n".printf(chroot, target_distro.dist_id);
+				else{
+					sh += "%s grub2-mkconfig -o /boot/grub2/grub.cfg \n".printf(chroot);
+				}
```

Comment:
`/boot/grub2/grub.cfg` is where it is for both EFI and non-EFI since Fedora 34 and RHEL/CentOS 9: https://fedoraproject.org/wiki/Changes/UnifyGrubConfig

Reply:
Also this won't work `dist_id` won't work for Redhat as the path uses `"redhat"` but `dist_id` will be `"rhel"`